hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -6664,6 +6664,8 @@ public String getDecomNodes() {
           node.getLeavingServiceStatus().getOutOfServiceOnlyReplicas())
           .put("underReplicateInOpenFiles",
           node.getLeavingServiceStatus().getUnderReplicatedInOpenFiles())
+          .put("decommissionDuration",
+              monotonicNow() - node.getLeavingServiceStatus().getStartTime())
```

Comment:
Hi, @zhangshuyan0 very nice feature. But i have a question here: how to judge DECOMMISSIONED and IN_MAINTENANCE state here? Looking forward to your reply.

Reply:
The `node` object here is from `decomNodeList`, which means that it is in decommissioning state.
https://github.com/apache/hadoop/blob/c35f31640ec4d74e85379d838e05c9c923d0cc77/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java#L6654-L6656

Reply:
Thanks a lot. @zhangshuyan0 .

Reply:
Please also check the failed unit tests if they are related to this PR, thanks a lot.

Reply:
It's not related. I can pass all of them in my local environment.